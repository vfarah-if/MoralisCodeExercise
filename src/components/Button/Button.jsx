import React from 'react';
import PropTypes from 'prop-types';

import './style/index.css';

const Button = ({ primary, backgroundColor, size, displayName, ...props }) => {
  const mode = primary ? 'button--primary' : 'button--secondary';
  return (
    <button
      type='button'
      className={['button', `button--${size}`, mode].join(' ')}
      style={backgroundColor && { backgroundColor }}
      {...props}
    >
      {displayName}
    </button>
  );
};

Button.propTypes = {
  primary: PropTypes.bool,
  backgroundColor: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  displayName: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  backgroundColor: null,
  primary: false,
  size: 'medium',
  onClick: undefined,
};

export default Button;
