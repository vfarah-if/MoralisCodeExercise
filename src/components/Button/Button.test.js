import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from './Button';

describe('button', () => {
  test('should create a button with a primary Button display name', () => {
    const { container } = render(
      <Button displayName='Button' primary={true} />
    );

    expect(screen.getByRole('button')).toHaveTextContent('Button');
    expect(container.querySelector('button')).toMatchSnapshot();
  });

  test('should create a button simple Button display name', () => {
    const renderResponse = render(<Button displayName='Button' />);

    expect(renderResponse.container.querySelector('button')).toMatchSnapshot();
  });

  test('should create a button large Button display name', () => {
    const renderResponse = render(<Button displayName='Button' size='large' />);

    expect(renderResponse.container.querySelector('button')).toMatchSnapshot();
  });

  test('should create a button small Button display name', () => {
    const renderResponse = render(<Button displayName='Button' size='small' />);

    expect(renderResponse.container.querySelector('button')).toMatchSnapshot();
  });
});
