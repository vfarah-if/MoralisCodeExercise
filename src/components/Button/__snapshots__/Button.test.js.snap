// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should create a button large Button display name 1`] = `
<button
  class="button button--large button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button simple Button display name 1`] = `
<button
  class="button button--medium button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button small Button display name 1`] = `
<button
  class="button button--small button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button with a primary Button display name 1`] = `
<button
  class="button button--medium button--primary"
  type="button"
>
  Button
</button>
`;
